internal/gateways/identity_api.go

```diff
@@ -0,0 +1,253 @@
+package gateways
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"math/big"
+	"net/http"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/rs/zerolog"
+)
+
+// IdentityVehicle represents a vehicle record returned by identity-api.
+type IdentityVehicle struct {
+	TokenID json.Number `json:"tokenId"`
+}
+
+// IdentityAPI defines methods used to interact with identity-api.
+type IdentityAPI interface {
+	VerifyDeveloperLicense(clientID string) (bool, int, error)
+	HasVehiclePermissions(vehicleTokenID string, devLicense []byte) (bool, error)
+	GetSharedVehicles(devLicense []byte) ([]IdentityVehicle, error)
+}
+
+type cachedPerm struct {
+	allowed bool
+	expires time.Time
+}
+
+type identityAPIService struct {
+	url        string
+	httpClient *http.Client
+	logger     zerolog.Logger
+	mu         sync.Mutex
+	permCache  map[string]cachedPerm
+}
+
+// NewIdentityAPIService creates a new IdentityAPI implementation.
+func NewIdentityAPIService(url string, logger zerolog.Logger) IdentityAPI {
+	return &identityAPIService{
+		url:        url,
+		httpClient: &http.Client{Timeout: 10 * time.Second},
```

Comment:
consider using the shared `http.NewClientWrapper` and then you can simplify your code in your below methods.

eg: `http.NewClientWrapper(settings.IdentityAPIURL.String(), "", 10*time.Second, nil, true)`
This will avoid you having to set "application/json" on every POST request below, and also having to specify the url every time.

Another advantage is you can just do this to query graphql endpoints:
`err := i.httpClient.GraphQLQuery("", query, &wrapper)`
where wrapper is the struct that represents the graphql json response your `result` variable line 86